app/models/affiliate.rb

```diff
@@ -739,8 +739,13 @@ def load_css_properties
   end
 
   def clear_existing_attachments
-    mobile_logo.clear if mobile_logo? && !mobile_logo.dirty? && mark_mobile_logo_for_deletion == '1'
-    header_tagline_logo.clear if header_tagline_logo? && !header_tagline_logo.dirty? && mark_header_tagline_logo_for_deletion == '1'
+    if mobile_logo? and !mobile_logo.dirty? and mark_mobile_logo_for_deletion == '1'
+      mobile_logo.clear
+    end
+
+    if header_tagline_logo? and !header_tagline_logo.dirty? and mark_header_tagline_logo_for_deletion == '1'
+      header_tagline_logo.clear
+    end
```

Comment:
Whoops - I should have clarified that we can keep the change from `and` to `&&`.

Reply:
Done.

I originally did this fix fairly mechanically, but now that I look and think, I'm minded to pull those two conditions out to their own predicates (`#need_to_clear_mobile_logo` and `#need_to_clear_header_tagline_logo`), just for the additional clarity. Objections?

Reply:
Let's just leave it. Additional refactoring is out of scope for this ticket.